app/models/task.py

```diff
@@ -1,5 +1,46 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
-
+from datetime import datetime
+from app import db
+from typing import Optional
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[datetime] = mapped_column(nullable=True, default=None)
+    # is_complete: Mapped[bool] = mapped_column(nullable=True) 
+    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id"), nullable=True)
+    goal = relationship("Goal", back_populates="tasks")
+    
+    
+    @property
+    def is_complete(self) -> bool:
+        return self.completed_at is not None
```

Comment:
Nice research into `@property` so you could reference it like an attribute and calculate the value whenever you need it.